### PR TITLE
Add support for `SignContainerIdentity` option

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
After the merge of https://github.com/sigstore/cosign/pull/2984, we now can use the new option to support it within the release-sdk as well.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added support for `SignContainerIdentity` option
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,9 +11,9 @@ require (
 	github.com/magefile/mage v1.15.0
 	github.com/maxbrunsfeld/counterfeiter/v6 v6.6.2
 	github.com/nozzle/throttler v0.0.0-20180817012639-2ea982251481
-	github.com/sigstore/cosign/v2 v2.0.3-0.20230517070756-fc61f43262cb
+	github.com/sigstore/cosign/v2 v2.0.3-0.20230614173616-4a2f67e796b9
```
Comment (saschagrunert) [inline go.mod:14]: We can either wait for a new release here or just go ahead as-is.

Comment (cpanato) [reply go.mod:14]: we are planning to cut a release soon, maybe today :)

File: go.mod [hunk 2]
```diff
@@ -11,9 +11,9 @@ require (
...
 	github.com/sigstore/rekor v1.2.1
-	github.com/sigstore/sigstore v1.6.5
+	github.com/sigstore/sigstore v1.6.6-0.20230602075123-c74dd4383337
```
Comment (cpanato) [inline go.mod:16]: i will cut a release for this one now !

Comment (cpanato) [reply go.mod:16]: https://github.com/sigstore/sigstore/releases/tag/v1.7.0

**Review by cpanato — APPROVED** [review]
Thanks

/lgtm

**Pull request merged**